spec/models/course/condition/assessment_spec.rb

```diff
@@ -0,0 +1,107 @@
+require 'rails_helper'
+
+RSpec.describe Course::Condition::Assessment, type: :model do
+  it { is_expected.to act_as(:condition).class_name(Course::Condition.name) }
+
+  let(:instance) { create(:instance) }
+  with_tenant(:instance) do
+    let(:course) { create(:course) }
+
+    describe '#title' do
+      let(:assessment) { create(:course_assessment_assessment, title: 'Dummy', course: course) }
+      subject { create(:course_condition_assessment, assessment: assessment) }
+
+      context 'when there is no minimum grade percentage' do
+        it 'returns the assessment title' do
+          expect(subject.title).to eq(assessment.title)
+        end
+      end
+
+      context 'when there is minimum grade percentage' do
+        it 'returns the assessment title with the minimum grade percentage' do
+          subject.minimum_grade_percentage = 33.333
+
+          expect(subject.title).
+            to eq(Course::Condition::Assessment.
+                    human_attribute_name('title.title',
+                                         assessment_title: assessment.title,
+                                         minimum_grade_percentage: '33.33%'))
+        end
+      end
+    end
+
+    describe '#satisfied_by?' do
+      let(:course_user) { create(:course_user) }
+      let(:assessment) do
+        # 10 questions, 2 marks each => maximum grade is 20
+        assessment = create(:assessment)
+        create_list(:course_assessment_question_multiple_response, 10, assessment: assessment)
```

Comment:
Do you need so many? Create only as many as necessary, this can slow down the specs.